### PR TITLE
[profiler] Expand Kineto platform support

### PR DESCRIPTION
**Pull request by ilia-cher**
Summary:
Expanding support to all builds

Test Plan:
CI

### CODE REVIEW

File: torch/csrc/autograd/profiler_kineto.h [hunk 1]
```diff
@@ -2,16 +2,13 @@
...
-#if !defined(__linux__) || defined(_WIN32) || defined(C10_MOBILE) || defined(__APPLE__) || defined(DISABLE_KINETO)
+#ifdef DISABLE_KINETO
 #undef USE_KINETO
 #endif
-#endif
 
 #ifdef USE_KINETO
 namespace libkineto {
-class TraceActivity;
+struct TraceActivity;
```
Comment (chaekit) [inline torch/csrc/autograd/profiler_kineto.h:11]: afaik this is basically the same for most cases. Is there any functional reason this is now fwd declared as a `struct`?

File: cmake/Dependencies.cmake [hunk 2]
```diff
@@ -1821,18 +1822,6 @@ list(APPEND Caffe2_DEPENDENCY_LIBS fmt::fmt-header-only)
 set(BUILD_SHARED_LIBS ${TEMP_BUILD_SHARED_LIBS} CACHE BOOL "Build shared libs" FORCE)
 
 # ---[ Kineto
-if(USE_KINETO AND INTERN_BUILD_MOBILE)
-  message(STATUS "Not using libkineto in a mobile build.")
```
Comment (chaekit) [inline cmake/Dependencies.cmake:1825]: nit: the message is confusing - it seems like we are expanding to Android/iOS which are "mobile". is `INTERN_BUILD_MOBILE` for FB internal mobile builds?

File: cmake/TorchConfig.cmake.in [hunk 3]
```diff
@@ -121,6 +125,10 @@ else()
   endif()
 
   append_torchlib_if_found(sleef asmjit)
+
+  if(@USE_KINETO@)
```
Comment (chaekit) [inline cmake/TorchConfig.cmake.in:129]: this seems like a duplicate. can be outside of the if-else block

File: cmake/Dependencies.cmake (hunk 2 above)
Comment (ilia-cher) [reply cmake/Dependencies.cmake:1825]: i'm removing this, we're expanding to all builds yes

File: torch/csrc/autograd/profiler_kineto.h (hunk 1 above)
Comment (ilia-cher) [reply torch/csrc/autograd/profiler_kineto.h:11]: there're compiler warnings that in some cases this might cause an issue, so i've corrected it

**Comment by chaekit** [comment]
lgtm

**Comment by ilia-cher** [comment]
will continue in https://github.com/pytorch/pytorch/pull/56323

**Pull request closed without merging**